src/core/io/socket.ts

```diff
@@ -411,7 +412,7 @@ export class Socket {
       this.controller.emitEvent(EventName.disconnected);
     }
 
-    this.controller.resume();
+    this.controller.pause();
```

Comment:
It's a big logical difference, care to explain?

Reply:
I'm not sure it's a big difference as starting to send messages when the socket is closed won't send anything, but the correct behaviour is to pause the communication.